File: Calculator_GUI/Delphi/Calculator.pas
unit Calculator;

interface

uses
  System.SysUtils, System.Types, System.UITypes, System.Classes, System.Variants,
  FMX.Types, FMX.Controls, FMX.Forms, FMX.Graphics, FMX.Dialogs, FMX.StdCtrls,
  FMX.Controls.Presentation;

type
  TCalculatorGUI = class(TForm)
    Del: TButton;
    C: TButton;
    Percent: TButton;
    Plus: TButton;
    Seven: TButton;
    Eight: TButton;
    Nine: TButton;
    Minus: TButton;
    Four: TButton;
    Five: TButton;
    Six: TButton;
    Multiply: TButton;
    One: TButton;
    Two: TButton;
    Three: TButton;
    Zero: TButton;
    Dot: TButton;
    Divide: TButton;
    Equals: TButton;
    Results: TLabel;
    Calc: TLabel;
    procedure DelClick(Sender: TObject);
    procedure ResultsClick(Sender: TObject);
    procedure CClick(Sender: TObject);
    procedure PercentClick(Sender: TObject);
    procedure PlusClick(Sender: TObject);
    procedure SevenClick(Sender: TObject);
    procedure EightClick(Sender: TObject);
    procedure NineClick(Sender: TObject);
    procedure MinusClick(Sender: TObject);
    procedure FourClick(Sender: TObject);
    procedure FiveClick(Sender: TObject);
    procedure SixClick(Sender: TObject);
    procedure MultiplyClick(Sender: TObject);
    procedure OneClick(Sender: TObject);
    procedure TwoClick(Sender: TObject);
    procedure ThreeClick(Sender: TObject);
    procedure DivideClick(Sender: TObject);
    procedure ZeroClick(Sender: TObject);
    procedure DotClick(Sender: TObject);
    procedure EqualsClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  CalculatorGUI: TCalculatorGUI;

implementation

{$R *.fmx}

procedure TCalculatorGUI.DelClick(Sender: TObject);
begin
     //
end;procedure TCalculatorGUI.ResultsClick(Sender: TObject);
begin
     //
end;
procedure TCalculatorGUI.CClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.DivideClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.DotClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.EightClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.EqualsClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.FiveClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.FourClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.MinusClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.MultiplyClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.NineClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.OneClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.PercentClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.PlusClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.SevenClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.SixClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.ThreeClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.TwoClick(Sender: TObject);
begin
     //
end;

procedure TCalculatorGUI.ZeroClick(Sender: TObject);
begin
     //
end;

end.

File: Calculator_GUI/Python/Calculator.py
import os
from delphifmx import *

class CalculatorGUI(Form):
    def __init__(self, owner):
        self.Del = None
        self.C = None
        self.Percent = None
        self.Plus = None
        self.Seven = None
        self.Eight = None
        self.Nine = None
        self.Minus = None
        self.Four = None
        self.Five = None
        self.Six = None
        self.Multiply = None
        self.One = None
        self.Two = None
        self.Three = None
        self.Zero = None
        self.Dot = None
        self.Divide = None
        self.Equals = None
        self.Calc = None
        self.Results = None
        self.LoadProps(os.path.join(os.path.dirname(os.path.abspath(__file__)), "Calculator.pyfmx"))
        self.Calculation = ""
        self.Results.Text = ""
        self.Calc.Text = ""

    def DelClick(self, Sender):
        self.Calculation = self.Calculation[:-1]
        self.Calc.Text = self.Calculation

    def CClick(self, Sender):
        self.Calculation = ""
        self.Calc.Text = self.Calculation
        self.Results.Text = ""

    def PercentClick(self, Sender):
        self.Calculation += "%"
        self.Calc.Text = self.Calculation

    def PlusClick(self, Sender):
        self.Calculation += "+"
        self.Calc.Text = self.Calculation

    def SevenClick(self, Sender):
        self.Calculation += "7"
        self.Calc.Text = self.Calculation

    def EightClick(self, Sender):
        self.Calculation += "8"
        self.Calc.Text = self.Calculation

    def NineClick(self, Sender):
        self.Calculation += "9"
        self.Calc.Text = self.Calculation

    def MinusClick(self, Sender):
        self.Calculation += "-"
        self.Calc.Text = self.Calculation

    def FourClick(self, Sender):
        self.Calculation += "4"
        self.Calc.Text = self.Calculation

    def FiveClick(self, Sender):
        self.Calculation += "5"
        self.Calc.Text = self.Calculation

    def SixClick(self, Sender):
        self.Calculation += "6"
        self.Calc.Text = self.Calculation

    def MultiplyClick(self, Sender):
        self.Calculation += "*"
        self.Calc.Text = self.Calculation

    def OneClick(self, Sender):
        self.Calculation += "1"
        self.Calc.Text = self.Calculation

    def TwoClick(self, Sender):
        self.Calculation += "2"
        self.Calc.Text = self.Calculation

    def ThreeClick(self, Sender):
        self.Calculation += "3"
        self.Calc.Text = self.Calculation

    def ZeroClick(self, Sender):
        self.Calculation += "0"
        self.Calc.Text = self.Calculation

    def DotClick(self, Sender):
        self.Calculation += "."
        self.Calc.Text = self.Calculation

    def DivideClick(self, Sender):
        self.Calculation += "/"
        self.Calc.Text = self.Calculation

    def EqualsClick(self, Sender):
        try:
            self.Results.Text = str(eval(self.Calculation))
        except:
            self.Results.Text = "Error"